activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -165,6 +165,50 @@ def attribute_names
         end
       end
 
+      def enforce_raw_sql_whitelist(args, whitelist: attribute_names_and_aliases) # :nodoc:
+        return if allow_unsafe_raw_sql == :enabled
+
+        unexpected = args.reject do |arg|
+          whitelist.include?(arg.to_s) ||
+            arg.kind_of?(Arel::Node) || arg.is_a?(Arel::Nodes::SqlLiteral)
+        end
+
+        return if unexpected.none?
+
+        if allow_unsafe_raw_sql == :deprecated
+          ActiveSupport::Deprecation.warn(
+            "Dangerous query method (method who'se arguments are used as raw " +
```

Comment:
The correct way to do a multiline string literal in Ruby is to end the line with a `\` like so:

```
"line 1 " \
"line 2"
````